split-webapp/split/src/components/TransactionFilterButton.js

```diff
@@ -0,0 +1,193 @@
+import React, { Component } from "react";
+import FilterListIcon from "@material-ui/icons/FilterList";
+import { Grid } from "@material-ui/core";
+import Modal from "@material-ui/core/Modal";
+import PropTypes from "prop-types";
+import styles from "./TransactionFilterButton.module.css";
+
+/* eslint jsx-a11y/no-onchange: 0 */
+class TransactionFilterButton extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      open: false,
+      sortField: "created_time",
+      sortOrder: "desc",
+      isPaid: "either",
+      payer: "",
+      payee: ""
+    };
+  }
+
+  render() {
+    const handleOpen = () => {
+      this.setState({ open: true });
+    };
+    const { sortField, sortOrder, isPaid, payer, payee } = this.state;
+
+    const handleSubmit = () => {
+      const { changeFilters } = this.props;
+      changeFilters(sortField, sortOrder, isPaid, payer, payee);
+      this.setState({ open: false });
+    };
+
+    const handleClose = () => {
+      this.setState({ open: false });
+    };
+
+    const { open } = this.state;
+
+    const handleChangeSortField = event => {
+      this.setState({ sortField: event.target.value });
+    };
+    const handleChangeSortOrder = event => {
+      this.setState({ sortOrder: event.target.value });
+    };
+    const handleChangeIsPaid = event => {
+      this.setState({ isPaid: event.target.value });
+    };
+    const handleChangePayer = event => {
+      this.setState({ payer: event.target.value });
+    };
+    const handleChangePayee = event => {
+      this.setState({ payee: event.target.value });
+    };
+    const handleClear = () => {
+      this.setState({
+        sortField: "created_time",
+        sortOrder: "desc",
+        isPaid: "either",
+        payer: "",
+        payee: ""
+      });
+    };
+
+    const modalStyles = {
+      position: "absolute",
+      top: "50%",
+      left: "50%",
+      transform: "translate(-50%, -50%) !important"
+    };
+
+    return (
+      <div>
+        <Grid container direction="row" alignItems="center">
+          <Grid item>
+            <FilterListIcon
+              className={styles.FilterTransactions}
+              onClick={handleOpen}
+            />
+          </Grid>
+          <Grid item>
+            <p className={styles.filterText}>Filter</p>
+          </Grid>
+        </Grid>
+        <Modal
+          open={open}
+          onClose={handleSubmit}
```

Comment:
Do you reckon clicking outside the modal should cancel the filter rather than applying it?

Reply:
for the onClose on the modal, if I don't have handleSubmit the button for submitting doesn't seem to work either for some reason

Reply:
Huh, interesting. By not working, is it not applying the filter, or not closing the dialog at all? It *feels* like it's working when I put `onClose={handleClose}`.

Reply:
I believe it doesn't apply the filters